integration-tests/conftest.py

```diff
@@ -1,10 +1,36 @@
 import pytest
+import subprocess
 
 
 @pytest.fixture(scope="session")
-def register_subman(external_candlepin, subman_session, test_config):
-    subman_session.register(
-        username=test_config.get("candlepin", "username"),
-        password=test_config.get("candlepin", "password"),
-    )
+def install_katello_rpm(test_config):
+    if "satellite" in test_config.environment:
+        # install katello rpm before register system against Satellite
+        cmd = "rpm -qa 'katello-ca-consumer*' | xargs --no-run-if-empty rpm -e"
+        ret = subprocess.check_call(cmd, shell=True)
+
+        satellite_hostname = test_config.get("candlepin", "host")
+        cmd = [
+            "rpm",
+            "-Uvh",
+            "http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % satellite_hostname,
+        ]
+        ret = subprocess.check_call(cmd)
+
+
+@pytest.fixture(scope="session")
+def register_subman(
+    external_candlepin, subman_session, test_config, install_katello_rpm
+):
+    environments = test_config.get("candlepin", "environments")
+    if "satellite" in environments[0]:
```

Comment:
`candlepin.environments` specifies the environments to use for registration, not what is the current dynaconf environment (yes, unfortunately the same word has different meanings...)

you need to pass `candlepin.environments` as `environments` parameter to the `register()`, either when using AKs or in both cases

regarding the dynaconf environment: please use `test_config.environment`, which returns the dynaconf environment currently in use -- exactly as done in `install_katello_rpm` :)

Reply:
> you need to pass `candlepin.environments` as `environments` parameter to the `register()`, either when using AKs or in both cases
> 

Pino, you mean here: https://github.com/ptoscano/pytest-client-tools/blob/main/pytest_client_tools/subscription_manager.py#L56 ?

For now, it not necessary to pass environment parameter to subman register(). From Archana's PR for conftest, we know that environment is not needed for her testing. And if I use activationkey and org for subman registration in insights-client+satellite testing, environment parameter is also not needed for me.

We are clear that Archana use username/password for prod_cert env, and use activationkey/org for prod_cert_satellite env, which are the easiest way for both testings. I know both the envs can use both username/password mode and activation key mode for sub-man registration. But I prefer not to change this code now. If someday, Archana has the requirement to use activationkey in the prod_cert env, we can change the code then.

Reply:
OK wrt not using environments for registration for now

OTOH, the two lines I mentioned use the wrong way to check for the dynaconf environment, and thus need to be fixed.